packages/plugin-compiler/src/main/kotlin/io/realm/compiler/RealmSchemaLoweringExtension.kt

```diff
@@ -224,6 +144,96 @@ class RealmSchemaLoweringExtension : IrGenerationExtension {
     }
 }
 
+/**
+ * Locate the list of schema class literals. This currently support them being defined in-place as var args, or in place
+ * using some of the factory collection methods like setOf and arrayOf, but we don't follow the code if defined further
+ * away than that.
+ *
+ * E.g. all of these are valid
+ * ```
+ * RealmConfiguration(schema = setOf(MyType::class))
+ * RealmConfiguration.Builder(schema = arrayOf(MyType::class)).build()
```

Comment:
this is no longer true since you removed the varargs from the `RealmConfiguration` public ctor ...
only `RealmConfiguration(schema = setOf(Sample::class))` mode is supported